src/components/button.tsx

```diff
@@ -59,6 +59,19 @@ const secondaryStyles = `
 [&.is-secondary:not(:focus)]:aria-disabled:hover:shadow-a8c-gray-5
 `.replace( /\n/g, ' ' );
 
+const tertiaryStyles = `
+[&.is-tertiary]:text-white
+[&.is-tertiary]:bg-gray-700
+[&.is-tertiary]:focus:bg-gray-600
+[&.is-tertiary]:focus:text-white
+[&.is-tertiary:not(.is-destructive,:disabled,[aria-disabled=true])]:hover:bg-gray-600
+[&.is-tertiary:not(.is-destructive,:disabled,[aria-disabled=true])]:hover:text-white
+[&.is-tertiary]:hover:bg-white
+[&.is-tertiary]:hover:text-white
+[&.is-tertiary]:disabled:bg-gray-700
+[&.is-tertiary]:disabled:text-a8c-gray-50
+`.replace( /\n/g, ' ' );
```

Comment:
I noted these new tertiary styles modified the existing usage of tertiary in the add site modal. We might consider consolidating button styles (i.e., ask the design team if we can reduce the number of button styles used) or rename this new style. WDYT? 

| Before | After |
| - | - |
| <img width="1012" alt="before" src="https://github.com/Automattic/studio/assets/438664/341fe794-9356-431f-9abb-e8ac6740c3e4"> | <img width="1012" alt="after" src="https://github.com/Automattic/studio/assets/438664/6d0e49b5-9339-4e8b-8ad9-56c560377255"> |